01.A Basic integration/README.md

```diff
@@ -0,0 +1,234 @@
+# 01 Basic interaction
+
+In this sample we will introduce our first React component. This component will hold the contacts array in its state and use it to render the table.
+
+## Let's get started
+
+First of all we'll add `react` and `react-dom` into our app libraries. We manage this with `npm` so we just need to do `npm install` and get this packages.
+
+```shell
+npm install --save react react-dom
+```
+
+Next we'll link this dependencies into our `index.html`:
+
+```html
+...
+</div>
+
+  <script src="./node_modules/jquery/dist/jquery.js"></script>
+  <script src="./node_modules/react/dist/react.js"></script>
+  <script src="./node_modules/react-dom/dist/react-dom.js"></script>
+
+  <script src="./assets/js/app/App.js"></script>
+...
+```
+
+We'll also remove the `table` element and replace it with a div with id *tableComponent* that will contain our React component.
+
+Then we'll remove`domUtils.js` helper as we only needed it to render the table. We will replace this with our React table component:
+
+```html
+...
+<main class="main">
+  <h4>My contacts</h4>
+  <div id="tableComponent"></div>
+</main>
+...
+```
+
+After that we will create a namespace called `components` inside our `App.js` to hold our React components:
+
+```javascript
+(function initializeApp(window) {
+  'use strict';
+
+  var App = {};
+  App.components = {};
+
+  window.App = App;
+})(window);
+```
+
+Let's write our React `TableComponent`, first we'll create a `components` folder and a `TableComponent.js` file inside it. Our folders tree should look like this:
+
+```
+.
+├── assets
+│   ├── css
+│   │   └── styles.css
+│   └── js
+│       ├── app
+│       │   ├── App.js
+│       │   ├── components
+│       │   │   └── TableComponent.js
+│       │   ├── modules
+│       │   │   └── contactsModule.js
+│       │   └── services
+│       │       └── contactsService.js
+│       └── index.js
+└── index.html
+```
+
+## Defining our table component
+
+Let's write our React `TableComponent`. This will need two dependencies, React to be created and App to be exposed so we'll define our component and wrap it in an _IIFE _(Immediately Invoked Function Expression) and pass this dependencies:
+
+```javascript
+// TableComponent.js
+(function initializeTableComponent(React, App) {
+  'use strict';
+
+  var TableComponent = function(props) {
+
+  };
+
+  App.components.TableComponent = TableComponent;
+})(React, window.App);
+```
+
+Our component will need a contacts array from props to render the table. Let's add some `propTypes` validation right below the `TableComponent` definition:
+
```

Comment:
It would be a good idea to add a remark here, since we are using ES5 we have to use the CreateElement approach (not very developer friendly), in sample 1.5 we port this sample to ES6 + JSX